internal/storage/vmbus/vmbus.go

```diff
@@ -0,0 +1,45 @@
+// +build linux
+
+package vmbus
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// WaitForVMBusGUIDPath waits for the vmbus device to exist at /sys/bus/vmbus/devices/<vmBusGUID>
+func WaitForVMBusGUIDPath(ctx context.Context, vmBusGUID string) (string, error) {
+	vmBusGUIDPath := fmt.Sprintf("/sys/bus/vmbus/devices/%s*", vmBusGUID)
```

Comment:
At this level of the package it will always be exactly `%s` right? And then its not a `filepath.Glob` its a `filepath.Stat` right?

Reply:
That's true, though keeping filepath.Glob allows for more code reuse with WaitForFileMatchingPattern. I can make a separate func for stat'ing the vmbus path if you'd prefer.

Reply:
But I am saying your func here is asking for the `vmBusGUID` which is not a `pattern` at all. I am ok with making it more generic if you like so that we dont need to wait for the various 2nd level dirs in say networking and PCI but then you probably want to make the vmbus.WaitForDevice(ctx, vmBusGUIDPattern string)` so someone can pass `<GUID>/net` and this code can wait for the entire sub tree. Up to you I guess how you would like to unify here.

Reply:
I made this change. It doesn't stop us from having to wait twice in the network scenario (since we need to wait for /sys/bus..../<guid>/net to exist and additionally for a target directory under that to exist), but it does cut down waiting for the pci case to two :)

Reply:
@jterry75 is this change good?